sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -92,7 +92,8 @@ public static int calculateBitSetWidthInBytes(int numFields) {
       Arrays.asList(new DataType[]{
         StringType,
         BinaryType,
-        IntervalType
+        IntervalType,
+        new DecimalType()
```

Comment:
This code seems to match default Decimal types (precision=10, scale=0) only.
We need to match all decimal types with valid precision/scale.